docs/user/tutorial-asgi.rst

```diff
@@ -0,0 +1,1090 @@
+implement the image collection, and the individual image resources (the code
+below should go into ``images.py``):
+
+.. code:: python
+
+    import aiofiles
+    import falcon
+
+
+    class Images:
+
+        def __init__(self, config, store):
+            self.config = config
+            self.store = store
+
+        async def on_get(self, req, resp):
+            resp.media = [image.serialize() for image in self.store.list_images()]
+
+        async def on_get_image(self, req, resp, image_id):
+            image = self.store.get(str(image_id))
+            resp.stream = await aiofiles.open(image.path, 'rb')
+            resp.content_type = falcon.MEDIA_JPEG
+
+        async def on_post(self, req, resp):
+            data = await req.stream.read()
+            image_id = str(self.config.uuid_generator())
+            image = await self.store.save(image_id, data)
+
+            resp.location = image.uri
+            resp.media = image.serialize()
+            resp.status = falcon.HTTP_201
+
+Since the first iteration on the ``Images`` class is quite lean, we opted for
+implementing two resources, image collection (which supports ``GET`` for
+listing the collection, and ``POST`` for uploading a new image) and single
+image (which supports ``GET`` for downloading the image), under one class
+employing responder name suffixes.
+
+If the application continues to grow in complexity, it might get worth to make
+the code cleaner by splitting classes to strictly represent one RESTful
+resource per class. See also: :ref:`recommended-route-layout`
+
+.. note::
+   Here, we serve the image by simply assigning an open ``aiofiles`` file to
+   :attr:`resp.stream <falcon.asgi.Response.stream>`.
+
+.. warning::
+   In production deployment, serving files directly from the web server, rather
+   than through the Falcon ASGI app, will likely be more efficient, and therefore
+   should be preferred. See also: :ref:`faq_static_files`
+
+Running Our Application
+-----------------------
+
+Let's refactor our ``app.py`` to allow ``create_app()``\ing whenever we need
+it, be it tests or the ASGI application module:
+
+.. code:: python
+
+    import falcon.asgi
+
+    from .config import Config
+    from .images import Images
+    from .store import Store
+
+
+    def create_app(config=None):
+        config = config or Config()
+        store = Store(config)
+        images = Images(config, store)
+
+        app = falcon.asgi.App()
+        app.add_route('/images', images)
+        app.add_route('/images/{image_id:uuid}.jpeg', images, suffix='image')
+
+        return app
+
+But how about route suffixes for the ``Images`` class?
+Here, we have to remember to map the single image resource to the
+``'/images/{image_id:uuid}.jpeg'`` URI template using the ``'image'``
+suffix in the respective :func:`add_route <falcon.asgi.App.add_route>` call.
+
+The ASGI application now resides in ``asgi.py``:
+
+.. Copy-paste under: examples/asgilook/asgilook/asgi.py
```

Comment:
we can use instead
```
.. literalinclude:: ../../examples/asgilook/asgilook/asgi.py
   :language: python
```

I've tried and it works as expected without configuring anything else in sphinx. This would also make the "check_copy_pasrte" superfluous

this applies to the other snipped as well

Reply:
This is embarrassing :man_facepalming: 

Does `literalinclude` figure out source code language automatically?
Anyway, I'll look into it. That sounds awesome if we can simplify this stuff.

Reply:
That's the second line, `: language python:`

Reply:
Doh... thanks!